### PR TITLE
Added UISearchBar+RASignalSupport

### PR DESCRIPTION
**Pull request by vasilenkoigor**
Related to [PR](https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3178)

### CODE REVIEW

**Comment by vasilenkoigor** [comment]
@ikesyo Sorry, I'm fixed failure build already :)

File: ReactiveObjC.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -318,6 +318,10 @@
 		A9B3161C1B3940AF0001CB9C /* RACUnit.h in Headers */ = {isa = PBXBuildFile; fileRef = D03764BE19EDA41200A782A9 /* RACUnit.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		A9B316341B394C7F0001CB9C /* RACCompoundDisposableProvider.d in Sources */ = {isa = PBXBuildFile; fileRef = D037646A19EDA41200A782A9 /* RACCompoundDisposableProvider.d */; };
 		A9B316351B394C7F0001CB9C /* RACSignalProvider.d in Sources */ = {isa = PBXBuildFile; fileRef = D03764A319EDA41200A782A9 /* RACSignalProvider.d */; };
+		B7528AE01D8F10DD00C2D281 /* UISearchBar+RACSignalSupport.h in Headers */ = {isa = PBXBuildFile; fileRef = B7528ADE1D8F10DD00C2D281 /* UISearchBar+RACSignalSupport.h */; };
+		B7528AE11D8F10DD00C2D281 /* UISearchBar+RACSignalSupport.h in Headers */ = {isa = PBXBuildFile; fileRef = B7528ADE1D8F10DD00C2D281 /* UISearchBar+RACSignalSupport.h */; };
```
Comment (ikesyo) [inline ReactiveObjC.xcodeproj/project.pbxproj]: You should make the headers public to be visible to outside of the framework.

File: ReactiveObjC/ReactiveObjC.h [hunk 2]
```diff
@@ -73,6 +73,7 @@ FOUNDATION_EXPORT const unsigned char ReactiveObjCVersionString[];
 	#import <ReactiveObjC/UITableViewHeaderFooterView+RACSignalSupport.h>
 	#import <ReactiveObjC/UITextField+RACSignalSupport.h>
 	#import <ReactiveObjC/UITextView+RACSignalSupport.h>
+	#import <ReactiveObjC/UISearchBar+RACSignalSupport.h>
```
Comment (ikesyo) [inline ReactiveObjC/ReactiveObjC.h]: Place this line alphabetically.

**Comment by ikesyo** [comment]
Could you add a unit test for this?

**Comment by vasilenkoigor** [comment]
@ikesyo sure! But tests of project is failure right now, because doesn't found `too` in Nimble :(

**Comment by ikesyo** [comment]
What is the `too` method? Nimble doesn't have that and we don't have the usage.

**Comment by vasilenkoigor** [comment]
@ikesyo yes, it's my mistake :)

**Comment by vasilenkoigor** [comment]
@ikesyo I'm add tests for my PR asap 👍

File: ReactiveObjC/UISearchBar+RACSignalSupport.m [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#import "RACSignal+Operations.h"
+#import <objc/runtime.h>
+
+@implementation UISearchBar (RACSignalSupport)
+
+static void RACUseDelegateProxy(UISearchBar *self) {
+    if (self.delegate == self.rac_delegateProxy) return;
+
+    self.rac_delegateProxy.rac_proxiedDelegate = self.delegate;
+    self.delegate = (id)self.rac_delegateProxy;
```
Comment (mdiep) [inline ReactiveObjC/UISearchBar+RACSignalSupport.m]: The indentation is this file should be tabs, not spaces, to match the project conventions.

File: ReactiveObjC/UISearchBar+RACSignalSupport.m [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+- (RACSignal *)rac_textSignal {
+    @weakify(self);
+    RACSignal *signal = [[[[RACSignal defer:^{
+        @strongify(self);
+        return [self.rac_delegateProxy signalForSelector:@selector(searchBar:textDidChange:)];
+    }] reduceEach:^(UISearchBar *searchBar, NSString *searchBarText) {
+        return searchBarText;
+    }] takeUntil:self.rac_willDeallocSignal] setNameWithFormat:@"%@ -rac_textSignal", RACDescription(self)];
```
Comment (mdiep) [inline ReactiveObjC/UISearchBar+RACSignalSupport.m]: This should be formatted in the RAC style, where each new operator starts on its own line, indented 1 tab stop.

```objc
   RACSignal *signal = [[[[RACSignal
        defer:^{
            @strongify(self);
            return [self.rac_delegateProxy signalForSelector:@selector(searchBar:textDidChange:)];
        }]
        reduceEach:^(UISearchBar *searchBar, NSString *searchBarText) {
            return searchBarText;
        }]
        takeUntil:self.rac_willDeallocSignal]
        setNameWithFormat:@"%@ -rac_textSignal", RACDescription(self)];
```

**Comment by Insofan** [comment]
How to implement
`self.searchBar.rac_textSignal`
my rac version is 2.5

**Comment by ikesyo** [comment]
Closing this since the forked repository is removed.

**Pull request closed without merging**